### PR TITLE
Implement IsSampled for OpenTelemetry bridgeSpanContext

### LINKED ISSUE #3532: Add `IsSampled` method to OpenTracing bridge
### Problem Statement

Bridge for OpenTracing does not expose whether span was picked for sampling.

### Proposed Solution

Add a method `bridgeSpanContext.IsSampled`.

### Additional Context

Ideally, this information would be exposed over OpenTracing interfaces but it's not (see https://github.com/opentracing/specification/issues/92 ). We work around it by invoking [Jaeger's `IsSampled`](https://github.com/jaegertracing/jaeger-client-go/blob/8d8e8fcfd04de42b8482476abac6a902fca47c18/span_context.go#L177-L181) directly. This information is used to optimize some of the critical paths.

I think it's not necessary to expose `bridgeSpanContext` all together. It would be enough to define a public `IsSampled` method that can be access by the few clients that want to have this information by casting it to a corresponding interface.

If you decide you are OK with the proposal, I should be able to make a PR for it.

### PR DESCRIPTION
**Pull request by pijusn**
This is a PR in response to issue https://github.com/open-telemetry/opentelemetry-go/issues/3532 .

`IsSampled` is not exposed by OpenTracing but its implementations did (at least some; sometimes called differently; Jaeger does). Cosidering that in OpenTracing this information was accessed by casting types, this implementation is no different - instances can be casted to an interface with expected methods in order to access it. Once moved to the bridge, eventually bridge can be dropped and this information would continue being exposed through the OpenTelemetry's span context.

Resolves #3532

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   - `Int64UpDownCounter` replaces the `syncint64.UpDownCounter`
   - `Int64Histogram` replaces the `syncint64.Histogram`
 - Add `NewTracerProvider` to `go.opentelemetry.io/otel/bridge/opentracing` to create `WrapperTracer` instances from a `TracerProvider`. (#3316)
+- Add `bridgetSpanContext.IsSampled` to `go.opentelemetry.io/otel/bridget/opentracing` to expose whether span is sampled or not.
```
Comment (MrAlias) [inline CHANGELOG.md:53]:
```suggestion
- Add `bridgetSpanContext.IsSampled` to `go.opentelemetry.io/otel/bridget/opentracing` to expose whether span is sampled or not. (#3570)
```

**Review by MrAlias — COMMENTED** [review]
Please update the [README](https://github.com/open-telemetry/opentelemetry-go/blob/7e5d903305cd315def2e350b58cedc1220bf7a68/bridge/opentracing/README.md) with a section on how the sampled state can be accessed.  Additionally, it might be worth adding an [example test](https://go.dev/blog/examples) that accesses this method.

**Comment by pijusn** [comment]
I'm not sure example test is worthwhile here. It would contain a lot of boilerplate just to set it up and then very trivial example. Also, there is no documentation for it to be attached to. I would skip it for this component.

If you do feel strongly about it, I can add it still. Let me know.

**Comment by pijusn** [comment]
@MrAlias could you see the latest changes? Would be much appreciated.

File: CHANGELOG.md (hunk 1 above)
Comment (MrAlias) [reply CHANGELOG.md:72]: This is not resolved.

File: CHANGELOG.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `Extrema` type to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
   This type is used to represent min/max values and still be able to distinguish unset and zero values. (#3487)
 - The `go.opentelemetry.io/otel/semconv/v1.17.0` package.
+- Add `bridgetSpanContext.IsSampled` to `go.opentelemetry.io/otel/bridget/opentracing` to expose whether span is sampled or not.
   The package contains semantic conventions from the `v1.17.0` version of the OpenTelemetry specification. (#3599)
```
Comment (MrAlias) [inline CHANGELOG.md:73]:
```suggestion
- The `go.opentelemetry.io/otel/semconv/v1.17.0` package.
  The package contains semantic conventions from the `v1.17.0` version of the OpenTelemetry specification. (#3599)
- Add `bridgetSpanContext.IsSampled` to `go.opentelemetry.io/otel/bridget/opentracing` to expose whether span is sampled or not.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -69,6 +69,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - The `Extrema` type to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
   This type is used to represent min/max values and still be able to distinguish unset and zero values. (#3487)
 - The `go.opentelemetry.io/otel/semconv/v1.17.0` package.
+- Add `bridgetSpanContext.IsSampled` to `go.opentelemetry.io/otel/bridget/opentracing` to expose whether span is sampled or not.
```
Comment (MrAlias) [inline CHANGELOG.md:72]: This needs to be moved to the unreleased section.

File: bridge/opentracing/README.md [hunk 4]
```diff
@@ -38,3 +38,24 @@ When you have started an OpenTracing Span, make sure the OpenTelemetry knows abo
 	// Propagate the otSpan to both OpenTracing and OpenTelemetry
 	// instrumentation by using the ctxWithOTAndOTelSpan context.
 ```
+
+## Extended Functionality
+
+While the bridge does not expose functionality that is not implemented by OpenTelemetry, it does expose some that is part of OpenTelemetry API and not OpenTracing API.
```
Comment (MrAlias) [inline bridge/opentracing/README.md:44]:
```suggestion
The bridge functionality can be extended beyond the OpenTracing API.
```

File: bridge/opentracing/README.md [hunk 5]
```diff
@@ -38,3 +38,24 @@ When you have started an OpenTracing Span, make sure the OpenTelemetry knows abo
...
+
+**`SpanContext.IsSampled`**
```
Comment (MrAlias) [inline bridge/opentracing/README.md:46]:
```suggestion
### `SpanContext.IsSampled`
```

File: bridge/opentracing/README.md [hunk 6]
```diff
@@ -38,3 +38,24 @@ When you have started an OpenTracing Span, make sure the OpenTelemetry knows abo
...
+
+Proxies underlying `trace.IsSampled` method (see [documentation](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext.IsSampled)). In order to use it, you have to cast it:
```
Comment (MrAlias) [inline bridge/opentracing/README.md:48]:
```suggestion
Return the underlying OpenTelemetry [`Span.IsSampled`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext.IsSampled) value by converting a `bridgeSpanContext`.
```

File: bridge/opentracing/README.md [hunk 7]
```diff
@@ -38,3 +38,24 @@ When you have started an OpenTracing Span, make sure the OpenTelemetry knows abo
...
+type samplable interface {
+	IsSampled() bool
+}
+
+var sc opentracing.SpanContext = ...
+if sc.(samplable).IsSampled() {
+	// Span is expected to be sampled.
+} else {
+	// Span will be discarded.
+}
```
Comment (MrAlias) [inline bridge/opentracing/README.md:60]:
```suggestion
if s, ok := sc.(samplable); ok && s.IsSampled() {
	// Do something with sc knowing it is sampled.
}
```

**Comment by pijusn** [comment]
Merged your suggestions, moved note to the unreleased, added PR ID to the note and merged the `main` branch into the PR. Should be up-to-date now.

**Review by MrAlias — APPROVED** [review]

**Comment by pijusn** [comment]
I merged `main` into this branch to resolve the conflict. Do you have any timeline when it could be merged (no merge rights over here, it will have to be you)? It conflicts every time something gets appended to the CHANGELOG.

**Comment by MrAlias** [comment]
This still needs another review from someone else before it is ready to merge.

**Review by dmathieu — APPROVED** [review]

**Comment by pijusn** [comment]
Thank you for reviewing the PR. I updated the branch to the latest `main`. It has no conflicts at the moment. I do not have the permissions required for the merge. I hope it gets merged soon 🤞🏻

**Pull request merged**